atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CellValuePutter.java

```diff
@@ -61,18 +70,22 @@ public CellValuePutter(
 
     void putWithOverriddenTimestamps(
             final String kvsMethodName, final TableReference tableRef, final Iterable<Map.Entry<Cell, Value>> values) {
-        put(kvsMethodName, tableRef, values, true);
+        putInternal(kvsMethodName, tableRef, values, Optional.of(timestampOverrideSupplier.getAsLong()));
     }
 
     void put(final String kvsMethodName, final TableReference tableRef, final Iterable<Map.Entry<Cell, Value>> values) {
-        put(kvsMethodName, tableRef, values, false);
+        putInternal(kvsMethodName, tableRef, values, Optional.empty());
     }
 
-    private void put(
+    void set(final String kvsMethodName, final TableReference tableRef, final Iterable<Map.Entry<Cell, Value>> values) {
+        putInternal(kvsMethodName, tableRef, values, Optional.of(SET_TIMESTAMP));
+    }
+
+    private void putInternal(
```

Comment:
It is internal, but might be useful to have docs to explain how `overrideTimestamp` works